vowpalwabbit/search.cc

```diff
@@ -2943,6 +2942,28 @@ base_learner* setup(options_i& options, vw& all)
   l.set_end_examples(end_examples);
   l.set_finish(search_finish);
   l.set_end_pass(end_pass);
+
+  // In search, tasks can define which label should be used. There isn't a great
+  // way to do this right now. However, currently the only usage is for cost
+  // sensitive. So we will check at this point if the label parser is either
+  // multiclass or cost sensitive. In any other case throw as it is not
+  // supported yet. TODO: improve the handling of tasks specifying label types.
+  if (all.p->lp.parse_label == COST_SENSITIVE::cs_label.parse_label)
```

Comment:
Hmm, it could be. But given there's only two branches and a default I don't think it's worth it. Especially because it would be switching on pointers which I think is a tad more surprising.